File: backend/extract_errors.js
const fs = require('fs');
const path = require('path');

const appExPattern = /AppException\.(badRequest|notFound|unauthorized|forbidden|internalServerError)\(\s*"(.*?)"\s*\)/g;
const apiResPattern = /ApiResponse\.error\(\s*".*?"\s*,\s*"(.*?)"\s*\)/g;

const strings = new Set();

function walkDir(dir) {
    const files = fs.readdirSync(dir);
    for (const file of files) {
        const fullPath = path.join(dir, file);
        if (fs.statSync(fullPath).isDirectory()) {
            walkDir(fullPath);
        } else if (file.endsWith('.java')) {
            const content = fs.readFileSync(fullPath, 'utf-8');
            let match;
            while ((match = appExPattern.exec(content)) !== null) {
                strings.add(match[2]);
            }
            while ((match = apiResPattern.exec(content)) !== null) {
                strings.add(match[1]);
            }
        }
    }
}

walkDir(path.join(__dirname, 'src'));
console.log(JSON.stringify(Array.from(strings), null, 2));

File: backend/extract_errors.py
import os
import re
import json

app_ex_pattern = re.compile(r'AppException\.(badRequest|notFound|unauthorized|forbidden|internalServerError)\(\s*"(.*?)"\s*\)')
api_res_pattern = re.compile(r'ApiResponse\.error\(\s*".*?"\s*,\s*"(.*?)"\s*\)')

strings = set()
for root, dirs, files in os.walk(r'C:\Users\Dell\Downloads\SWP_Project-develop (1)\SWP_Project-develop\backend\src'):
    for file in files:
        if file.endswith('.java'):
            path = os.path.join(root, file)
            with open(path, 'r', encoding='utf-8') as f:
                content = f.read()
                for match in app_ex_pattern.finditer(content):
                    strings.add(match.group(2))
                for match in api_res_pattern.finditer(content):
                    strings.add(match.group(1))

print(json.dumps(list(strings), indent=2))
